kernel/src/main/java/org/kframework/kompile/Kompile.java

```diff
@@ -297,9 +298,37 @@ private void checkDefinition(Definition parsedDef, Set<String> excludedModuleTag
         scala.collection.Set<Module> modules = parsedDef.modules();
         Module mainModule = parsedDef.mainModule();
         Option<Module> kModule = parsedDef.getModule("K");
+        definitionChecks(stream(modules).collect(Collectors.toSet()));
         structuralChecks(modules, mainModule, kModule, excludedModuleTags, true);
     }
 
+    // checks that are not verified in the prover
+    public void definitionChecks(Set<Module> modules) {
+        modules.forEach(m -> stream(m.localSentences()).forEach(s -> {
+            // Check that the `claim` keyword is not used in the definition.
+            if (s instanceof Claim)
+                errors.add(KEMException.compilerError("Claims are not allowed in the definition.", s));
+        }));
+    }
```

Comment:
Add a test to `k-distribution/tests/regression-new/checks` which exercises this case.